client/question/services/ParentPageServiceSpec.js

```diff
@@ -0,0 +1,47 @@
+// Copyright 2017 The TIE Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for ParentPageService.
+ */
+
+describe('ParentPageService', function() {
+  var ParentPageService;
+
+  beforeEach(module('tie'));
+  var setParentPage = function(parentPageOrigin) {
+    module('tieConfig', function($provide) {
+      $provide.constant('PARENT_PAGE_ORIGIN', parentPageOrigin);
+    });
+  };
+
+  describe('getParentPageOrigin', function() {
```

Comment:
no tests for sendMessage? Maybe see if you can mock window.parent.

Reply:
Added tests to verify postMessage.